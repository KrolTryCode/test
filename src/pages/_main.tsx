import { Stack } from '@mui/material';
import { createFileRoute, Outlet } from '@tanstack/react-router';

import { Header } from '~/routing/layout/header/header.component';
import { Submenu } from '~/routing/layout/submenu/submenu.component';
import { useNotifyAboutNewNotification } from '~/use-cases/use-notify-about-new-notification.hook';

export const Route = createFileRoute('/_main')({
  component: MainLayout,
});

function MainLayout() {
  useNotifyAboutNewNotification();

  return (
    <Stack width={'100vw'} height={'100dvh'}>
      <Header />
      <Submenu />
      <Stack component={'main'} zIndex={1} flex={1} overflow={'auto'} gap={1} padding={1}>
        <Outlet />
      </Stack>
    </Stack>
  );
}
