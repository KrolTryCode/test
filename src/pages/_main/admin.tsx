import { createFileRoute, redirect } from '@tanstack/react-router';

export const Route = createFileRoute('/_main/admin')({
  staticData: {
    accessBy: ['ADMIN'],
    title: 'NAVIGATION.ADMIN',
    order: 1,
  },
  beforeLoad: ({ location }) => {
    if (location.pathname === '/admin' || location.pathname === '/admin/') {
      return redirect({ to: '/admin/users' });
    }
  },
});
