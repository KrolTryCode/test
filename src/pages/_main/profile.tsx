import { createFileRoute } from '@tanstack/react-router';

import { getCurrentUserQueryOptions } from '~/api/queries/users/get-current-user.query';
import { ProfileForm } from '~/components/forms/profile/profile-form';
import { ProfileAvatar } from '~/components/user-avatar/editable-user-avatar.component';
import {
  UserProfileContent,
  UserProfileHeader,
  UserProfileLayout,
} from '~/components/user-profile/user-profile.style';
import { getFullName } from '~/components/user-profile/user-profile.utils';
import { useProfile } from '~/use-cases/profile.hook';

export const Route = createFileRoute('/_main/profile')({
  component: Profile,
  staticData: {
    title: 'USER.LABEL',
  },
  loader: async ({ context }) => {
    const { user } = await context.queryClient.fetchQuery(getCurrentUserQueryOptions());
    const fullName = getFullName(user?.firstName, user?.lastName, user?.surName);
    context.title = fullName;
    return { user: { ...user, fullName } };
  },
});

export function Profile() {
  const { user } = Route.useLoaderData();
  const { handleChangePassword, handleUpdateUser } = useProfile({ id: user.id, email: user.email });

  return (
    <UserProfileLayout>
      <UserProfileHeader userName={user.fullName} />
      <UserProfileContent>
        <ProfileAvatar
          userId={user.id ?? ''}
          firstName={user?.firstName}
          lastName={user?.lastName}
          surName={user?.surName}
        />
        <ProfileForm
          data={user}
          handleUpdateUser={handleUpdateUser}
          isCurrent
          handleChangePassword={handleChangePassword}
        />
      </UserProfileContent>
    </UserProfileLayout>
  );
}
