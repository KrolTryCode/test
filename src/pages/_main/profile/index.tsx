import { createFileRoute } from '@tanstack/react-router';

import { ProfileForm } from '~/components/user-profile/user-profile-form/profile-form.component';
import {
  UserProfileContent,
  UserProfileHeader,
  UserProfileLayout,
} from '~/components/user-profile/user-profile.style';

import { ProfileAvatar } from './profile-avatar/profile-avatar.component';
import { useProfile } from './profile.hook';

export const Route = createFileRoute('/_main/profile/')({
  component: Profile,
  staticData: {
    title: 'USER.LABEL',
  },
});

export function Profile() {
  const { user, isUserLoading, handleChangePassword, handleUpdateUser } = useProfile();

  return (
    <UserProfileLayout>
      <UserProfileHeader userName={user.fullName} />
      <UserProfileContent>
        <ProfileAvatar
          userId={user.id ?? ''}
          firstName={user?.firstName}
          lastName={user?.lastName}
          surName={user?.surName}
        />
        <ProfileForm
          data={user}
          isLoading={isUserLoading}
          handleUpdateUser={handleUpdateUser}
          isCurrent
          handleChangePassword={handleChangePassword}
        />
      </UserProfileContent>
    </UserProfileLayout>
  );
}
