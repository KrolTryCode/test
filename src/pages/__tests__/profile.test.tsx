import { screen } from '@testing-library/react';
import { it, describe, expect } from 'vitest';

import { mockedParams } from 'tests/__mocks__/@tanstack/react-router.mock';
import { renderComponent } from 'tests/utils/render-component';
import { Profile } from '~/pages/_main/profile';

describe('Profile', () => {
  beforeAll(() => {
    mockedParams.mockReturnValue({ userId: 'test-admin-id-2c963f66afa6' });
    renderComponent(<Profile />, { route: '/' });
  });

  it('renders the profile', () => {
    const container = screen.getByTestId('appContainer');
    expect(container).toMatchSnapshot('profile');
  });
});
