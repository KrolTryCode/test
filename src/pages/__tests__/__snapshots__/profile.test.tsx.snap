// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Profile > renders the profile > profile 1`] = `
<div
  data-testid="appContainer"
>
  <div
    class="MuiStack-root css-1sahg7c-MuiStack-root"
  >
    <div
      class="MuiStack-root css-1qtjl8b-MuiStack-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h3 css-1tom75q-MuiTypography-root"
      />
    </div>
    <div
      class="MuiStack-root css-13ebx6i-MuiStack-root"
    >
      <div
        class="MuiStack-root profile-avatar css-1fs4ojd-MuiStack-root"
        variant="circular"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1dpsd08-MuiAvatar-root"
          color="primary"
          size="150"
          title=""
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1rr90df-MuiTypography-root"
          />
        </div>
        <div
          class="MuiStack-root css-36i21p-MuiStack-root"
          variant="circular"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-8jlasq-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            title="Загрузить фото профиля"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-uks7c6-MuiSvgIcon-root"
              data-testid="PhotoCameraIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="3.2"
              />
              <path
                d="M9 2 7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5"
              />
            </svg>
          </button>
        </div>
      </div>
      <form
        class="MuiStack-root css-1ymhfon-MuiStack-root"
        role="form"
      >
        <fieldset
          class="MuiStack-root css-106nt1m-MuiStack-root"
        >
          <legend
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-196qjet-MuiTypography-root"
          >
            <div
              class="MuiStack-root css-13tplj0-MuiStack-root"
            >
              Персональные данные
            </div>
          </legend>
          <fieldset
            class="MuiStack-root css-1javh08-MuiStack-root"
          >
            <div
              class="MuiFormControl-root css-csmh00-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1hud5yu-MuiFormLabel-root"
              >
                Фамилия
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk css-11c5b5w-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiBox-root css-eal7pa"
              >
                <div
                  class="MuiBox-root css-0"
                  data-mui-internal-clone-element="true"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-qh4m2g-MuiFormControl-root-MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1mqbila-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        aria-readonly="false"
                        autocomplete="off"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1l16dx2-MuiInputBase-input-MuiOutlinedInput-input"
                        id=":r1:"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-w4cd9x"
                        >
                          <span
                            class="notranslate"
                          >
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root css-csmh00-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1hud5yu-MuiFormLabel-root"
              >
                Имя
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk css-11c5b5w-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiBox-root css-eal7pa"
              >
                <div
                  class="MuiBox-root css-0"
                  data-mui-internal-clone-element="true"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-qh4m2g-MuiFormControl-root-MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1mqbila-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        aria-readonly="false"
                        autocomplete="off"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1l16dx2-MuiInputBase-input-MuiOutlinedInput-input"
                        id=":r3:"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-w4cd9x"
                        >
                          <span
                            class="notranslate"
                          >
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root css-csmh00-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1hud5yu-MuiFormLabel-root"
              >
                Отчество
              </label>
              <div
                class="MuiBox-root css-eal7pa"
              >
                <div
                  class="MuiBox-root css-0"
                  data-mui-internal-clone-element="true"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-qh4m2g-MuiFormControl-root-MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1mqbila-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        aria-readonly="false"
                        autocomplete="off"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1l16dx2-MuiInputBase-input-MuiOutlinedInput-input"
                        id=":r5:"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-w4cd9x"
                        >
                          <span
                            class="notranslate"
                          >
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </fieldset>
          <fieldset
            class="MuiStack-root css-1javh08-MuiStack-root"
          >
            <div
              class="MuiFormControl-root css-csmh00-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1hud5yu-MuiFormLabel-root"
              >
                Предприятие
              </label>
              <div
                class="MuiBox-root css-eal7pa"
              >
                <div
                  class="MuiBox-root css-0"
                  data-mui-internal-clone-element="true"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-qh4m2g-MuiFormControl-root-MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1mqbila-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        aria-readonly="false"
                        autocomplete="off"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1l16dx2-MuiInputBase-input-MuiOutlinedInput-input"
                        id=":r7:"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-w4cd9x"
                        >
                          <span
                            class="notranslate"
                          >
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root css-csmh00-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1hud5yu-MuiFormLabel-root"
              >
                Подразделение
              </label>
              <div
                class="MuiBox-root css-eal7pa"
              >
                <div
                  class="MuiBox-root css-0"
                  data-mui-internal-clone-element="true"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-qh4m2g-MuiFormControl-root-MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1mqbila-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        aria-readonly="false"
                        autocomplete="off"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1l16dx2-MuiInputBase-input-MuiOutlinedInput-input"
                        id=":r9:"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-w4cd9x"
                        >
                          <span
                            class="notranslate"
                          >
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root css-csmh00-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1hud5yu-MuiFormLabel-root"
              >
                Должность
              </label>
              <div
                class="MuiBox-root css-eal7pa"
              >
                <div
                  class="MuiBox-root css-0"
                  data-mui-internal-clone-element="true"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-qh4m2g-MuiFormControl-root-MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1mqbila-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        aria-readonly="false"
                        autocomplete="off"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1l16dx2-MuiInputBase-input-MuiOutlinedInput-input"
                        id=":rb:"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-w4cd9x"
                        >
                          <span
                            class="notranslate"
                          >
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </fieldset>
        </fieldset>
        <fieldset
          class="MuiStack-root css-106nt1m-MuiStack-root"
        >
          <legend
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-196qjet-MuiTypography-root"
          >
            <div
              class="MuiStack-root css-13tplj0-MuiStack-root"
            >
              Контакты
            </div>
          </legend>
          <fieldset
            class="MuiStack-root css-1javh08-MuiStack-root"
          >
            <div
              class="MuiFormControl-root css-csmh00-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1hud5yu-MuiFormLabel-root"
              >
                E-mail
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk css-11c5b5w-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiBox-root css-eal7pa"
              >
                <div
                  class="MuiBox-root css-0"
                  data-mui-internal-clone-element="true"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-qh4m2g-MuiFormControl-root-MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1mqbila-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        aria-readonly="false"
                        autocomplete="email"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1l16dx2-MuiInputBase-input-MuiOutlinedInput-input"
                        id=":rd:"
                        type="email"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-w4cd9x"
                        >
                          <span
                            class="notranslate"
                          >
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </fieldset>
          <fieldset
            class="MuiStack-root css-1javh08-MuiStack-root"
          >
            <div
              class="MuiFormControl-root css-csmh00-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1hud5yu-MuiFormLabel-root"
              >
                Номер телефона
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk css-11c5b5w-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiBox-root css-eal7pa"
              >
                <div
                  class="MuiBox-root css-0"
                  data-mui-internal-clone-element="true"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root MuiTelInput-TextField css-qh4m2g-MuiFormControl-root-MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-clwuqp-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <div
                        class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1be8vlw-MuiInputAdornment-root"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                        <button
                          aria-expanded="false"
                          aria-haspopup="listbox"
                          aria-label="Select country"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiTelInput-IconButton css-1drddrs-MuiButtonBase-root-MuiIconButton-root"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiTelInput-Flag css-70qvj9"
                            data-testid="RU"
                          >
                            <div
                              class="fi fi-ru MuiBox-root css-1tr8hmv"
                            />
                          </span>
                        </button>
                      </div>
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1wj1ldm-MuiInputBase-input-MuiOutlinedInput-input"
                        id=":rf:"
                        type="tel"
                        value="+7"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-w4cd9x"
                        >
                          <span
                            class="notranslate"
                          >
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </fieldset>
        </fieldset>
        <fieldset
          class="MuiStack-root css-106nt1m-MuiStack-root"
        >
          <legend
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-196qjet-MuiTypography-root"
          >
            <div
              class="MuiStack-root css-13tplj0-MuiStack-root"
            >
              Системные данные
            </div>
          </legend>
          <fieldset
            class="MuiStack-root css-1javh08-MuiStack-root"
          >
            <div
              class="MuiFormControl-root css-csmh00-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1hud5yu-MuiFormLabel-root"
              >
                Дата регистрации
              </label>
              <div
                class="MuiBox-root css-eal7pa"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-el2kki-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd Mui-readOnly MuiInputBase-readOnly css-i9swye-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd Mui-readOnly MuiInputBase-readOnly css-7rlme6-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":rg:"
                      inputmode="text"
                      placeholder="ДД.ММ.ГГГГ чч:мм"
                      readonly=""
                      type="text"
                      value=""
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-k0bb7s-MuiInputAdornment-root"
                    >
                      <button
                        aria-label="Выберите дату"
                        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-edgeEnd MuiIconButton-sizeSmall css-1e1k4ac-MuiButtonBase-root-MuiIconButton-root"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                          data-testid="CalendarIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                          />
                        </svg>
                      </button>
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-w4cd9x"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root css-csmh00-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1hud5yu-MuiFormLabel-root"
              >
                Дата последнего входа
              </label>
              <div
                class="MuiBox-root css-eal7pa"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-el2kki-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd Mui-readOnly MuiInputBase-readOnly css-i9swye-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd Mui-readOnly MuiInputBase-readOnly css-7rlme6-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":ri:"
                      inputmode="text"
                      placeholder="ДД.ММ.ГГГГ чч:мм"
                      readonly=""
                      type="text"
                      value=""
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-k0bb7s-MuiInputAdornment-root"
                    >
                      <button
                        aria-label="Выберите дату"
                        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-edgeEnd MuiIconButton-sizeSmall css-1e1k4ac-MuiButtonBase-root-MuiIconButton-root"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                          data-testid="CalendarIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                          />
                        </svg>
                      </button>
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-w4cd9x"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root css-csmh00-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1hud5yu-MuiFormLabel-root"
              >
                IP-адрес последнего входа
              </label>
              <div
                class="MuiBox-root css-eal7pa"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-qh4m2g-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1mqbila-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      aria-readonly="true"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1l16dx2-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":rk:"
                      readonly=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-w4cd9x"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root css-csmh00-MuiFormControl-root"
              hidden=""
            >
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1hud5yu-MuiFormLabel-root"
              >
                Состояние
              </label>
              <div
                class="MuiBox-root css-eal7pa"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-qh4m2g-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1mqbila-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      aria-readonly="true"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1l16dx2-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":rl:"
                      readonly=""
                      type="text"
                      value="Нет данных"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-w4cd9x"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </fieldset>
        </fieldset>
        <div
          class="MuiStack-root css-1s8nmkv-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiLoadingButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-1g6sjdt-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
            id=":rm:"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiLoadingButton-label css-27yv9x-MuiLoadingButton-label"
            >
              Изменить пароль
            </span>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-13icm9f-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
            id=":rn:"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiLoadingButton-label css-27yv9x-MuiLoadingButton-label"
            >
              Сохранить
            </span>
          </button>
        </div>
      </form>
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-evrlh3-MuiBackdrop-root"
        style="opacity: 0; visibility: hidden;"
      >
        <div
          class="MuiStack-root css-19rwzzo-MuiStack-root"
        >
          <span
            class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-v4hvoy-MuiCircularProgress-root"
            role="progressbar"
            style="width: 40px; height: 40px;"
          >
            <svg
              class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate MuiCircularProgress-circleDisableShrink css-d3g403-MuiCircularProgress-circle"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1466jvw-MuiTypography-root"
          >
            Загрузка...
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
